thorium/response.py

```diff
@@ -75,15 +75,33 @@ def get_response_data(self):
 
     def _sort(self):
         if self.sort:
-            reverse = self._check_and_strip_first_char()
+            self.meta['sort'] = self.sort
             sort_by = self.sort.split(',')
-            for field in sort_by:
+            for field in reversed(sort_by):
+                field, reverse = self._check_and_strip_first_char(field)
                 if not hasattr(self.resources[0], field):
                     raise errors.BadRequestError(
                         'Cannot sort by field `{}`. It does not exist in the '
                         'resource.'.format(field)
                     )
-            self.resources.sort(key=attrgetter(*sort_by), reverse=reverse)
+                elif (isinstance(getattr(self.resources[0], field), dict) or
+                      isinstance(getattr(self.resources[0], field), list) or
```

Comment:
make this isinstance thing a separate function that accepts a resource and a type. Further, can you not refer to the class rather than doing that ugly index[0] thing?